### DISCUSSION TITLE
Group all renovate dependancies and lock file into one single MR

### DISCUSSION THREAD

**Discussion by juancarlosjr97** [renovatebot/renovate#22630] [Request Help] [answered]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

v35.112.1

### If you're self-hosting Renovate, select which platform you are using.

gitlab.com

### Was this something which used to work for you, and then stopped?

I never saw this working

### Wanted end result.

We would like to have all changes update dependencies and lock file maintenance into a single MR.

### What you tried so far.

We have self-hosted runners on GitLab and to avoid exhausting them, we have added the configuration as follows

```
groupName: "all dependencies",
separateMajorMinor: false,
groupSlug: "renovate-dependencies",
packageRules: [
  {
    matchPackagePatterns: ["*"],
    groupName: "all dependencies",
    groupSlug: "renovate-dependencies",
  },
],
lockFileMaintenance: {
  enabled: true,
  schedule: null,
},
```

To avoid creating multiple PRs all together as just one PR.

### Relevant debug logs

<details><summary>Logs</summary>

```
Copy/paste the relevant log(s) here, between the starting and ending backticks
```

</details>

**Comment by rarkins** [comment]
Lock file maintenance in the same PR is not advised and will be deprecated in future

**Reply by juancarlosjr97** [reply]
Ok, so there is not way?

**Reply by rarkins** [reply]
It can be fine if you use matchUpdateTypes=lockFileMaintenance

**Reply by juancarlosjr97** [reply]
Would that group everything into a single MR? I just want to be grouped but if it is going to be deprecated, and it is not advised, then perhaps not doing it is the best way.

**Reply by juancarlosjr97** [reply]
I see I can put everything into the configuration such as "major, minor, patch, pin, pinDigest, digest, lockFileMaintenance, rollback, bump, replacement" and let the configuration handle it :)

**Reply by viceice** [reply, marked as answer]
lockfile maintenance and replacements can no longer be grouped in renovate v43
- https://github.com/renovatebot/renovate/pull/40758
- https://github.com/renovatebot/renovate/pull/40781

**Comment by TiAlRo** [comment]
We would have liked to keep the possibility to group lock file maintenance and other updates in a single PR.

First, two PR at the same time lead to race conditions in the pipeline. One pipeline deploys a test deployment and before it can be tested, the other pipeline uninstalls the test deployment. Ofc this is not a clean pipeline setup which we should handle better anyway.

Second, even with clean pipeline setup, we would like to have auto-merge after a succeeding pipeline. I doubt this is possible when lock file maintenance and other updates are separated, but I have to test it.

**Reply by viceice** [reply]
no, because grouping them causes different troubles. that's the reason we did it.
